api/incubator/src/test/java/io/opentelemetry/api/incubator/propagation/CaseInsensitiveMapTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.incubator.propagation;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class CaseInsensitiveMapTest {
+
+  @Test
+  void createByConstructor() {
+    Map<String, String> map = new HashMap<>();
+    map.put("Key1", "test");
+    map.put("Key2", "test2");
+
+    CaseInsensitiveMap caseInsensitiveMap = new CaseInsensitiveMap(map);
+
+    for (Map.Entry<String, String> caseEntry : caseInsensitiveMap.entrySet()) {
+      String k = caseEntry.getKey();
+      String v = caseEntry.getValue();
+      boolean result = false;
+      for (Map.Entry<String, String> mapEntry : map.entrySet()) {
+        String k1 = mapEntry.getKey();
+        String v1 = mapEntry.getValue();
+
+        String keyLowerCase = caseInsensitiveMap.getKeyLowerCase(k1);
```

Comment:
I would just call `k1.toLowerCase()` rather than exposing a method on the implementation like this.
Same on the line below.